graylog2-web-interface/src/components/indices/rotation/TimeBasedSizeOptimizingStrategyConfiguration.tsx

```diff
@@ -0,0 +1,127 @@
+  index_lifetime_hard: string,
+  index_lifetime_soft: string,
+  type: string,
+}
+
+type Props = {
+  config: TimeBasedSizeOptimizingStrategyConfig,
+  updateConfig: (config: Omit<TimeBasedSizeOptimizingStrategyConfig, 'type'>) => void,
+}
+
+export const durationToRoundedDays = (duration: string) => {
+  return Math.round(moment.duration(duration).asDays());
+};
+
+const _validationLimit = (durationInDays, rotationLimit) => {
+  return durationInDays <= durationToRoundedDays(rotationLimit);
+};
+
+const TimeBasedSizeOptimizingStrategy = ({ config: { index_lifetime_hard, index_lifetime_soft }, updateConfig }: Props) => {
+  const [indexLifetimeHard, setIndexLifetimeHard] = useState(index_lifetime_hard);
+  const [indexLifetimeSoft, setIndexLifetimeSoft] = useState(index_lifetime_soft);
+  const maxRotationPeriod = useMaxIndexRotationLimit();
+
+  const _isValidPeriod = useCallback((duration: string) => {
+    const checkInDays = durationToRoundedDays(duration);
+
+    return checkInDays >= 1 && (
+      maxRotationPeriod ? _validationLimit(checkInDays, maxRotationPeriod) : true
+    );
+  }, [maxRotationPeriod]);
+
+  const formatDuration = useCallback((duration: string) => {
+    const maxRotationPeriodErrorMessage = maxRotationPeriod ? ` and max ${durationToRoundedDays(maxRotationPeriod)} days` : '';
+
+    return _isValidPeriod(duration) ? `${durationToRoundedDays(duration)} days` : `invalid (min 1 day${maxRotationPeriodErrorMessage})`;
+  }, [_isValidPeriod, maxRotationPeriod]);
+
+  const onLifetimeSoftChange = (event: React.ChangeEvent<HTMLOptionElement>): void => {
+    const inputValue = getValueFromInput(event.target);
+
+    setIndexLifetimeSoft(inputValue);
+
+    if (_isValidPeriod(inputValue)) {
+      updateConfig({ index_lifetime_hard: indexLifetimeHard, index_lifetime_soft: inputValue });
+    }
+  };
+
+  const onLifetimeHardChange = (event: React.ChangeEvent<HTMLOptionElement>): void => {
+    const inputValue = getValueFromInput(event.target);
+
+    setIndexLifetimeHard(inputValue);
+
+    if (_isValidPeriod(inputValue)) {
+      updateConfig({ index_lifetime_soft: indexLifetimeSoft, index_lifetime_hard: inputValue });
+    }
+  };
+
+  const validationState = (duration: string) => {
+    if (_isValidPeriod(duration)) {
+      return null;
+    }
+
+    return 'error';
+  };
+
+  const maxRotationPeriodHelpText = maxRotationPeriod ? ` The max rotation period is set to ${durationToRoundedDays(maxRotationPeriod)} by Administrator.` : '';
+
+  return (
+    <div>
+      <Input id="rotation-index-lifetime-soft"
+             type="text"
+             labelClassName="col-sm-3"
+             wrapperClassName="col-sm-9"
+             label="Soft lifetime (ISO8601 Duration)"
+             onChange={onLifetimeSoftChange}
+             value={indexLifetimeSoft}
+             help={`The minimum time an index gets written to before it is rotated. (i.e. "P1D" for 1 day, "PT6H" for 6 hours).${maxRotationPeriodHelpText}`}
```

Comment:
That's not how it works.
index-lifetime-soft is the minimum time time we want to keep the data available for this index set.
We won't be retaining it before that time has passed.

Lifetime hard is the maximum time we want to keep the data available for this index set. (mostly for GDPR reasons)

The rotation happens daily (for now) and thus the lifetimes should be only given as a multiple of days.

Furthermore, we don't allow the hard and soft lifetime to be identical. they need to be at least one day apart.

Reply:
Got it. Thank you for clarifying. I will update it accordingly.